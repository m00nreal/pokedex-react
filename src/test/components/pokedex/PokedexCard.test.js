import PokeCard from "../../../components/pokedex/PokedexCard";
import {mount, shallow} from "enzyme";
import React from "react";
import {CardImg, Spinner} from "reactstrap";

describe('Tests in <PokedexCard/>', function () {

    it('should match snapshot', function () {
        const init = {
            id: 1,
            image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
            name: 'bulbasaur',
            abilities: ['overgrow, chlorophyll'],
            height: 70,
            weight: 69,
            types: ['grass', 'poison'],
            loading: false
        }

        const wrapper = shallow(<PokeCard {...init}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render spinner if there is no info yet', function () {
        const init = {
            id: 0,
            image: '',
            name: '',
            abilities: [],
            height: 0,
            weight: 0,
            types: [],
            loading: true
        }

        const wrapper = shallow(<PokeCard {...init}/>);

        expect(wrapper.contains(<Spinner color="primary"/>)).toBe(true);
    })

    it('should set a default image if there is no image', function () {
        const init = {
            id: 1,
            image: '',
            name: 'bulbasaur',
            abilities: ['overgrow, chlorophyll'],
            height: 70,
            weight: 69,
            types: ['grass', 'poison'],
        }

        const wrapper = mount(<PokeCard {...init} loading={false}/>);
        expect(wrapper.contains(<CardImg className="align-self-center" width="50%" src="/pokedex-react/assets/images/noimg.png" alt={init.name}/>)).toBe(true);
    })

});