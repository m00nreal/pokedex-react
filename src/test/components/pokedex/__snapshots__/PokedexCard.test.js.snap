// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PokedexCard/> should match snapshot 1`] = `
<Card
  className="m-auto animate__animated animate__fadeIn"
  tag="div"
>
  <CardHeader
    className="text-center font-weight-bolder"
    tag="div"
  >
    Bulbasaur
     
    <span
      className="card-header__id small d-block  "
    >
      # 
      1
    </span>
  </CardHeader>
  <CardBody
    className="d-flex flex-column"
    tag="div"
  >
    <CardImg
      alt="bulbasaur"
      className="align-self-center"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      tag="img"
      width="50%"
    />
    <div>
      <li
        className="list-item"
      >
        Height: 
        <span>
          7
           m
        </span>
      </li>
      <li
        className="list-item"
      >
        Weight: 
        6.9
         kg
      </li>
      <div
        className="abilities"
      >
        <div
          className="abilities-header"
        >
          Abilities
        </div>
        <div
          className="abilities-content"
        >
          overgrow, chlorophyll
        </div>
      </div>
      <li
        className="list-item"
      >
        <Badge
          className="mr-1 p-1"
          color="secondary"
          key="grass"
          pill={false}
          style={
            Object {
              "background": "#95de64",
            }
          }
          tag="span"
        >
          grass
        </Badge>
        <Badge
          className="mr-1 p-1"
          color="secondary"
          key="poison"
          pill={false}
          style={
            Object {
              "background": "#3f6600",
            }
          }
          tag="span"
        >
          poison
        </Badge>
      </li>
    </div>
  </CardBody>
</Card>
`;
